pkg/resources/kafka/kafka.go

```diff
@@ -417,41 +426,93 @@ func arePodsAlreadyDeleted(pods []corev1.Pod, log logr.Logger) bool {
 	}
 	return true
 }
+func (r *Reconciler) getClientPasswordKeyAndUser() (string, string, error) {
+	var clientPass, CN string
+	if r.KafkaCluster.Spec.ListenersConfig.ClientSSLCertSecretName != "" || r.KafkaCluster.Spec.ListenersConfig.SSLSecrets != nil {
+		clientNamespacedName := types.NamespacedName{Name: fmt.Sprintf(pkicommon.BrokerControllerTemplate, r.KafkaCluster.Name), Namespace: r.KafkaCluster.Namespace}
+		if r.KafkaCluster.Spec.ListenersConfig.ClientSSLCertSecretName != "" {
+			clientNamespacedName = types.NamespacedName{Name: r.KafkaCluster.Spec.ListenersConfig.ClientSSLCertSecretName, Namespace: r.KafkaCluster.Namespace}
+		}
+		clientSecret := &corev1.Secret{}
+		if err := r.Client.Get(context.TODO(), clientNamespacedName, clientSecret); err != nil {
+			if apierrors.IsNotFound(err) {
+				return "", "", errorfactory.New(errorfactory.ResourceNotReady{}, err, "client secret not ready")
+			}
+			return "", "", errors.WrapIfWithDetails(err, "failed to get client secret")
+		}
+		cert, err := certutil.DecodeCertificate(clientSecret.Data[corev1.TLSCertKey])
+		if err != nil {
+			return "", "", errors.WrapIfWithDetails(err, "failed to decode client certificate")
+		}
+		CN = cert.Subject.String()
+		clientPass = string(clientSecret.Data[v1alpha1.PasswordKey])
+	}
+	return clientPass, CN, nil
+}
 
-func (r *Reconciler) getServerAndClientDetails() (string, string, []string, error) {
-	if r.KafkaCluster.Spec.ListenersConfig.SSLSecrets == nil {
-		return "", "", []string{}, nil
+func getListenerSSLCertSecret(client client.Reader, commonSpec v1beta1.CommonListenerSpec, clusterName string, clusterNamespace string) (*corev1.Secret, error) {
```

Comment:
Instead of adding the client and the cluster related things to the parameter list why dont we make this function to a method?

Reply:
If we use method here these parameters will be passed as hidden. I wouldn't add this func into the reconcile structure because it's only a helper function, also It doesn't change any data on the reconciler structure.